docs/native-modules-ios.md

```diff
@@ -1,442 +1,466 @@
-{
-  NSArray *events = ...
-  if (events) {
-    resolve(events);
-  } else {
-    NSError *error = ...
-    reject(@"no_events", @"There were no events", error);
-  }
+This Javascript file also becomes a good location for you to add any JavaScript side functionality. For example, if you use a type system like TypeScript you can add type annotations for your native module here. While React Native does not yet support Native to JS type safety, with these type annotations, all your JS code will be type safe. These annotations will also make it easier for you to switch to type-safe native modules down the line. Below is an example of adding type safety to the Calendar Module:
+
+```jsx
+/**
+* This exposes the native CalendarModule module as a JS module. This has a
+* function 'createCalendarEvent' which takes the following parameters:
+*
+* 1. String name: A string representing the name of the event
+* 2. String location: A string representing the location of the event
+*/
+import { NativeModules } from 'react-native';
+const { CalendarModule } = NativeModules
+interface CalendarInterface {
+   createCalendarEvent(name: string, location: string): void;
 }
+export default CalendarModule as CalendarInterface;
 ```
 
-The JavaScript counterpart of this method returns a Promise. This means you can use the `await` keyword within an async function to call it and wait for its result:
+In your other Javascript files you can access the native module and invoke its method like this:
 
 ```jsx
-const updateEvents = async () => {
-  try {
-    var events = await CalendarManager.findEvents();
+import NativeCalendarModule from './NativeCalendarModule';
+NativeCalendarModule.createCalendarEvent(‘foo’, ‘bar’);
+```
 
-    this.setState({ events });
-  } catch (e) {
-    console.error(e);
-  }
-};
+> Note this assumes that the place you are importing `CalendarModule` is in the same hierarchy as `CalendarModule.js`. Please update the relative import as necessary.
 
-updateEvents();
-```
+### Argument Types
 
-## Threading
+When a native module method is invoked in Javascript, React Native converts the arguments from JS objects to their Objective-C/Swift object analogues. So for example, if your Objective-C Native Module method accepts a NSNumber, in JS you need to call the method with a number. React Native will handle the conversion for you. Below is a list of the argument types supported for native module methods and the JavaScript equivalents they map to.
 
-The native module should not have any assumptions about what thread it is being called on. React Native invokes native modules methods on a separate serial GCD queue, but this is an implementation detail and might change. The `- (dispatch_queue_t)methodQueue` method allows the native module to specify which queue its methods should be run on. For example, if it needs to use a main-thread-only iOS API, it should specify this via:
+- string -> NSString \*
```

Comment:
Can you convert this list into the table? I think this will improve the readability.

Also what's the purpose of asterisk at the end of some entries?

Reply:
Updated to tables. Not sure where that asterisk came from, I had copied over from the original docs.

Reply:
It looks better, thank you! 

One more question, about for example `NSArray` - there are two very similar rows for this type, can this be simplified by putting in JS column `?Array | Array` or just `?Array`? (This hint might be wrong since I'm not that experienced with RN conversions)

<img width="607" alt="Screenshot 2020-11-28 145722" src="https://user-images.githubusercontent.com/719641/100517309-0ab7d600-318a-11eb-94f7-3c4c8c175763.png">

Reply:
@Simek followed up with the table params here: https://github.com/facebook/react-native-website/pull/2373